### PR TITLE
riot-desktop: use yarn2nix-moretea

### PR DESCRIPTION
**Pull request by ghost**
Now that we have yarn2nix-moretea in the tree, we can use that instead of the vendored version that was introduced when yarn2nix was removed previously. Also add an update script.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: pkgs/applications/networking/instant-messengers/riot/riot-desktop.nix [hunk 1]
```diff
@@ -1,10 +1,8 @@
-{ pkgs, stdenv, fetchFromGitHub, makeWrapper, makeDesktopItem, electron_5, riot-web }:
+{ pkgs, stdenv, fetchFromGitHub, makeWrapper, makeDesktopItem, electron_5, riot-web, yarn2nix-moretea }:
 
 # Note for maintainers:
 # Versions of `riot-web` and `riot-desktop` should be kept in sync.
```
Comment (pacien) [inline pkgs/applications/networking/instant-messengers/riot/riot-desktop.nix]: Could you add the following note for maintainers?

```suggestion
# * versions of `riot-web` and `riot-desktop` should be kept in sync.
# * the Yarn dependency expression must be updated with `./update-riot-desktop.sh <git release tag>`
```

File: pkgs/applications/networking/instant-messengers/riot/update-riot-desktop.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+ if [ "$#" -ne 1 ] || [[ "$1" == -* ]]; then
+  echo "Regenerates the Yarn dependency lock files for the riot-desktop package."
+  echo "Usage: $0 <git release tag>"
+  exit 1
+fi
+
+ RIOT_WEB_SRC="https://raw.githubusercontent.com/vector-im/riot-web/$1"
+
+ wget "$RIOT_WEB_SRC/electron_app/package.json" -O riot-desktop-package.json
```
Comment (pacien) [inline pkgs/applications/networking/instant-messengers/riot/update-riot-desktop.sh]: Nitpick: weird indent at lines 4, 6, 12 and 14.

**Review by pacien — APPROVED** [review]
Looks good to me, thanks!

File: pkgs/applications/networking/instant-messengers/riot/update-riot-desktop.sh (hunk 2 above)
Comment (petabyteboy) [reply pkgs/applications/networking/instant-messengers/riot/update-riot-desktop.sh]: Fail while pasting, thanks for noticing

File: pkgs/applications/networking/instant-messengers/riot/riot-desktop.nix (hunk 1 above)
Comment (petabyteboy) [reply pkgs/applications/networking/instant-messengers/riot/riot-desktop.nix]: Done

File: pkgs/applications/networking/instant-messengers/riot/update-riot-desktop.sh [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+	echo "Usage: $0 <git release tag>"
+	exit 1
+fi
+
+RIOT_WEB_SRC="https://raw.githubusercontent.com/vector-im/riot-web/$1"
+
+wget "$RIOT_WEB_SRC/electron_app/package.json" -O riot-desktop-package.json
+wget "$RIOT_WEB_SRC/electron_app/yarn.lock" -O riot-desktop-yarndeps.lock
+yarn2nix --lockfile=riot-desktop-yarndeps.lock > riot-desktop-yarndeps.nix
+rm riot-desktop-yarndeps.lock
```
[the remark this answers is not in the document]
Comment (pacien) [reply pkgs/applications/networking/instant-messengers/riot/update-riot-desktop.sh]: It's used to generate `riot-desktop-yarndeps.nix` and isn't necessary afterwards.

File: pkgs/applications/networking/instant-messengers/riot/riot-desktop.nix [hunk 4]
```diff
@@ -1,9 +1,8 @@
-{ pkgs, stdenv, fetchFromGitHub, makeWrapper, makeDesktopItem, electron_5, riot-web }:
+{ pkgs, stdenv, fetchFromGitHub, makeWrapper, makeDesktopItem, electron_5, riot-web, yarn2nix-moretea }:
 
-# Note for maintainers:
-# Versions of `riot-web` and `riot-desktop` should be kept in sync.
-
-with (import ./yarn2nix.nix { inherit pkgs; });
+# Notes for maintainers:
+# * versions of `riot-web` and `riot-desktop` should be kept in sync.
+# * the Yarn dependency expression must be updated with `./update-riot-desktop.sh <git release tag>`
```
Comment (pacien) [inline pkgs/applications/networking/instant-messengers/riot/riot-desktop.nix]: Could you also remove the comments at lines 23-24 and 27-28?

Comment (petabyteboy) [reply pkgs/applications/networking/instant-messengers/riot/riot-desktop.nix]: Sure

**Review by worldofpeace — APPROVED** [review]

**Pull request merged**